plugins/Goals/VisitorDetails.php

```diff
@@ -90,7 +92,15 @@ protected function queryGoalConversionsForVisits($idVisits)
 					AND log_conversion.idgoal > 0
                 ORDER BY log_conversion.idvisit, log_conversion.server_time ASC
 			";
-        $conversions = $this->getDb()->fetchAll($sql);
+
+        $sql = DbHelper::addMaxExecutionTimeHintToQuery($sql, $this->getLiveQueryMaxExecutionTime());
+
+        try {
+            $conversions = $this->getDb()->fetchAll($sql);
+        } catch (\Exception $e) {
+            Model::handleMaxExecutionTimeError($this->getDb(), $e, '', Date::now(), Date::now(), null, 0, ['sql' => $sql]);
+            throw $e;
+        }
```

Comment:
```suggestion
        $dateNowStart = Date::now();
        try {
            $conversions = $this->getDb()->fetchAll($sql);
        } catch (\Exception $e) {
            Model::handleMaxExecutionTimeError($this->getDb(), $e, '', $dateNowStart, Date::now(), null, 0, ['sql' => $sql]);
            throw $e;
        }
```

Reply:
Apologies, I know it's still in draft, I'll be logging off for the week so thought might be still worth mentioning.